lib/expeditor/command.rb

```diff
@@ -14,41 +14,50 @@ def initialize(opts = {}, &block)
       @service = opts.fetch(:service, Expeditor::Services.default)
       @timeout = opts[:timeout]
       @dependencies = opts.fetch(:dependencies, [])
-      @normal_future = initial_normal(&block)
-      @fallback_var = nil
+
+      @normal_future = nil
       @retryable_options = Concurrent::IVar.new
+      @normal_block = block
+      @fallback_block = nil
+      @ivar = Concurrent::IVar.new
     end
 
-    def start
-      if not started?
-        @dependencies.each(&:start)
+    # @param current_thread [Boolean] Execute the task on current thread(blocking)
+    def start(current_thread: false)
+      unless started?
+        if current_thread
+          prepare(Concurrent::ImmediateExecutor.new)
+        else
+          prepare
+        end
         @normal_future.safe_execute
       end
       self
     end
 
     # Equivalent to retryable gem options
     def start_with_retry(retryable_options = {})
-      if not started?
+      unless started?
+        current_thread = retryable_options.delete(:current_thread)
```

Comment:
`current_thread` is not a member of `retryable_options`, so this variable name is a little confusing. 
I think you should rename it, for example `options`.